aws/database/opensearch/addresses/scripts/clear_unused_indexes.py

```diff
@@ -0,0 +1,46 @@
+from aws.database.opensearch.client.elasticsearch_client import LocalElasticsearchClient
+from utils.confirm import confirm
+
+
+def find_aliased_indexes(es_client: LocalElasticsearchClient) -> list[str]:
+    instance_aliases = es_client.es_instance.indices.get_alias(index='*')
+    aliased_indexes: list[str] = []
+    for index_name, res in instance_aliases.items():
+        index_aliases: dict = res.get("aliases")
+        if index_aliases.keys():
+            aliased_indexes.append(index_name)
+    return aliased_indexes
+
+
+def storage_for_index(es_client: LocalElasticsearchClient, index: str) -> int:
+    stats = es_client.es_instance.indices.stats(index=index, metric='store')
+    size_in_bytes = stats['indices'][index]['total']['store']['size_in_bytes']
+    return size_in_bytes
+
+
+def main():
+    print("\n\n === ES Start == \n\n")
+
+    es_client = LocalElasticsearchClient(None)
+
+    all_indexes = es_client.list_all_indices()
+
+    indexes_to_preserve = find_aliased_indexes(es_client)
+    if not confirm(f"Preserving indexes: {indexes_to_preserve}"):
+        return
+
+    indexes_to_clear = [
+        index for index in all_indexes
+        if index not in indexes_to_preserve
+        and "address" in index
+    ]
```

Comment:
A code comment, or a comment under this script's Readme _(if there's one)_, explaining why non-aliased indexes are being cleared may be useful for future maintainers.

As not having an alias typically simply means, that there's no alternative name to reference by, not necessarily that: "It can't be referenced", or "It isn't used, or can't be used".

Reply:
Added a module docstring instead to not clutter the repo. Good point, for sure. The elasticsearch setup with addresses is far from conventional

https://github.com/LBHackney-IT/mtfh-scripts/pull/33/commits/2b43c16d265535b843ca00633a57ae4655b39308